python/examples/sorting-permutation.py

```diff
@@ -0,0 +1,73 @@
+import sys
+import os
+import shutil
+import segyio
+
+
+def pathjoin( prefix, path):
+    directory, base = os.path.split(path)
+    pref, suff = os.path.splitext(base)
+    filename = pref + '-' + prefix + suff
+    return os.path.join(directory, filename)
+
+# This script reverse the traces for a given file, by iline, xline and offset in
+# all combinations of the three.
+
+def main():
+    if len(sys.argv) < 2:
+        sys.exit("Usage: {} [source-file] [destination-file]".format(sys.argv[0]))
+
+    sourcefile = sys.argv[1]
+    destfile = sys.argv[2] if len(sys.argv) > 2 else sourcefile
+
+
+    for suf in ['dec-il-xl-off', 'dec-il-inc-xl-off', 'dec-xl-inc-il-off',
+                'dec-il-xl-inc-off', 'dec-il-off-inc-xl', 'dec-xl-off-inc-il',
+                'dec-off-inc-il-xl']:
+        fname = pathjoin(suf, destfile)
+        shutil.copyfile(sourcefile, fname)
+
+        with segyio.open(sourcefile) as src, segyio.open(fname, 'r+') as dst:
+            for i in range(1 + src.ext_headers):
+                dst.text[i] = src.text[i]
+
+            dst.bin = src.bin
+            dst.trace = src.trace
+            dst.header = src.header
+
+    with segyio.open(sourcefile) as src, segyio.open(pathjoin("dec-il-xl-off", destfile), 'r+') as dst:
```

Comment:
This is too long and unwieldy for my taste, and I think you even benefit readability-wise from doing it in two blocks.